sgl-kernel/csrc/cpu/norm.cpp

```diff
@@ -304,24 +304,33 @@ void fused_rmsnorm_gated_kernel_impl(
 
 template <typename scalar_t>
 void fused_add_layernorm_kernel_impl(
-    scalar_t* __restrict__ input,
+    scalar_t* __restrict__ output,
+    const scalar_t* __restrict__ input,
     scalar_t* __restrict__ residual,
     const scalar_t* __restrict__ weight,
     float* __restrict__ buffer,
     int64_t batch_size,
+    int64_t seq_len,
     int64_t hidden_size,
     int64_t input_strideN,
     float eps = 1e-5) {
   using bVec = at::vec::Vectorized<scalar_t>;
   using fVec = at::vec::Vectorized<float>;
+  TORCH_CHECK(
+      !((residual != nullptr) && (buffer == nullptr)),
+      "fused_add_layernorm_kernel_impl with a residual input requires a buffer.");
```

Comment:
usually we don't do checks like this:

* check at the entry function. residual can be set to be optional, and check whether it has_value() for your scenarios.
* skip buffer check. it is defined by yourself.